### PR TITLE
New original gamemode: carp ops

### PR DESCRIPTION
**Pull request by Archanial**
## About The Pull Request

The gamemode is based on nuke ops but everyone is carp.
Carps can activate pinpointers, open doors, activate buttons and recover disk from objects.
Clicking at airlock and waiting will emag it.
Calling in war will add additional health and damage to whole squad.
They are faster, healthier and more syndicate than normal carps.
They also start with medhuds and internal radio.

Carps can carry disky in mouth and activate the nuke.

## Changelog
:cl:
add: New original gamemode - card ops.
/:cl:

### CODE REVIEW

File: _maps/_basemap.dm [hunk 1]
```diff
@@ -1,4 +1,4 @@
-//#define LOWMEMORYMODE //uncomment this to load centcom and runtime station and thats it.
+#define LOWMEMORYMODE //uncomment this to load centcom and runtime station and thats it.
```
Comment (Archanial) [inline _maps/_basemap.dm]:
```suggestion
//#define LOWMEMORYMODE //uncomment this to load centcom and runtime station and thats it.
```

File: code/modules/mob/living/simple_animal/hostile/carp.dm [hunk 2]
```diff
@@ -229,4 +235,66 @@
...
+		B.attack_hand(src)
+		return
+	if(istype(target, /mob/living/carbon))
+		var/mob/living/carbon/L = target
+		if(L.stat == DEAD)
+			to_chat(src, "<span class='warning'>You start consuming the body to replenish your strenght.</span>")
+			if(do_after(src, 3 SECONDS, FALSE, target))
+				for(var/player in GLOB.player_list)
+					to_chat(player, "<span class='userdanger'>The [L.real_name] just got fished! What a [pick("disgrace", "failure", "looser")]!</span>")
+				L.gib(TRUE)
```
Comment (zeskorion) [inline code/modules/mob/living/simple_animal/hostile/carp.dm:276]: should remove nuke disk first

Comment (Archanial) [reply code/modules/mob/living/simple_animal/hostile/carp.dm:276]: gib doesn't destroy nuke disk

File: html/antagtips/le_fishe.html [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+	<h1>You are a Fishy Nuclear Operative!</h1>
+	<img src="fishe.png"/>
+	<p>The syndicate has provided you with a radio uplink a few implants and a pinpointer.</p>
+	<p><img src="nuke.png"/>You must detonate the nuclear bomb from your ship on Space Station 13. To do that, you must get the nuclear disk.<img src="disk.png"/></p>
+	<p>If you notice the disk is inside a contents of something, just click on it and wait.</p>
+	<p>To activate anything you would need just click on it. You can carry a disk in your mouth and activate nuke when hitting it while having disky in mouth.</p>
+	<p>You hack open airlocks with your mouth!</p>
+	<p>You can heal by eating dead bodies. Additionally everyone will be notified of your target getting fished! What a disgrace!</p>
+	<p>Click on the ID to copy its access to your internal fish id!</p>
+	<img src="https://i.kym-cdn.com/entries/icons/facebook/000/034/880/Untitled-1.jpg"/>
```
Comment (Ivniinvi) [inline html/antagtips/le_fishe.html:12]: Facebook link?

File: code/modules/antagonists/nukeop/fishop.dm [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+	name = "Fish Operative Leader"
+	roundend_category = "fish operatives"
+	antagpanel_category = "FishOp"
+	nukeop_outfit = /datum/outfit/fishop/leader
+
+/datum/antagonist/nukeop/fishop/admin_add(datum/mind/new_owner,mob/admin)
+	new_owner.assigned_role = "Fish Operative"
+	new_owner.add_antag_datum(src)
+	message_admins("[key_name_admin(admin)] has clown op'ed [key_name_admin(new_owner)].")
+	log_admin("[key_name(admin)] has clown op'ed [key_name(new_owner)].")
```
Comment (Ivniinvi) [inline code/modules/antagonists/nukeop/fishop.dm]: clown op

Comment (Archanial) [reply code/modules/antagonists/nukeop/fishop.dm]: 🙄

Comment (Archanial) [reply code/modules/antagonists/nukeop/fishop.dm]:
```suggestion
	log_admin("[key_name(admin)] has fish op'ed [key_name(new_owner)].")
```

Comment (Archanial) [inline code/modules/antagonists/nukeop/fishop.dm]:
```suggestion
	message_admins("[key_name_admin(admin)] has fish op'ed [key_name_admin(new_owner)].")
	log_admin("[key_name(admin)] has fish op'ed [key_name(new_owner)].")
```

File: html/antagtips/le_fishe.html (hunk 3 above)
Comment (Archanial) [reply html/antagtips/le_fishe.html:12]: le fishe

File: code/game/gamemodes/nuclear/fish_nuclear.dm [hunk 5]
```diff
@@ -0,0 +1,143 @@
+/datum/game_mode/nuclear/fishy
+	name = "fishy nuclear emergency"
+	config_tag = "fish_nuclear"
+	report_type = "fish_nuclear"
+	false_report_weight = 10
+	required_players = 30 // 30 players - 7 players to be the nuke ops = 23 players remaining
+	required_enemies = 8
+	recommended_enemies = 5
```
Comment (Archanial) [inline code/game/gamemodes/nuclear/fish_nuclear.dm]:
```suggestion
	required_enemies = 4
	recommended_enemies = 8
```

File: code/modules/mob/living/simple_animal/hostile/carp.dm [hunk 6]
```diff
@@ -229,4 +235,76 @@
...
+				health = min(maxHealth, health + heal)
+				bruteloss -= heal
+				fireloss -= heal
```
Comment (Archanial) [inline code/modules/mob/living/simple_animal/hostile/carp.dm]:
```suggestion
				bruteloss = min(0, bruteloss  - heal)
				fireloss = min(0, fireloss - heal)
```

**Comment by ike709** [comment]
Nice CL typo

**Comment by williamcll** [comment]
Huh, I think I've seen this on another server before.

**Comment by Programs-The-Station** [comment]
Edit: I have been informed that it always spawns the teams even if it isn't the game mode.
~Round 37065: Fishy Ops Failed to spawn antags it looks like. Failed back to Traitor Gamemode with no Fishy antags on round end report~

**Comment by PepegaSandwich** [comment]
do it

**Comment by Archanial** [comment]
Do what

**Comment by Crossedfall** [comment]
Was a good meme. Removed the TM and cleaned up the config.

**Pull request closed without merging**